### PR TITLE
added node.js week3 homework

### PR DESCRIPTION
**Pull request by STEP0HEN**

### CODE REVIEW

File: nodejs/week3/node.js-week3-Homework/calculator/index.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+const express = require("express");
+const calculator = express();
+const port = 3000;
+calculator.use(express.json());
+calculator.use(express.urlencoded({ extended: true }));
+
+// Calculator using :method (path) parameter:
+calculator.get("/calculator/:method", (request, response) => {
+	const queryParams = Object.values(request.query);
+	const numsMerged = [].concat.apply([], queryParams);
```
Comment (OlkaA) [inline nodejs/week3/node.js-week3-Homework/calculator/index.js]: I believe numsMerged variable is not necessary. 
const queryParams is itself an array, so you can do .map method on it.

**Review by OlkaA — APPROVED** [review]
Very good job with calculator and meal sharing.
The code is well structured and easy to read. Well done!
 
You could also move all calculation (in the image below) in warmup to separate function, 
to avoid repetition of the code in get and post method.
image

**Comment by STEP0HEN** [comment]
@OlkaA  Thank you so much for the feedback. 
I implemented the changes.

On line 9-10 I did that because the request.query.params are key : value pairs and when I console.log typeof(queryParams) I got an object

**Comment by OlkaA** [comment]
Hi, I think the typeof an array is an object. 
In JavaScript, arrays are technically objects; just with special behaviours and abilities.
The correct way to check if something is an actual array, using Array.isArray(yourArray) syntaxes.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray

**Comment by STEP0HEN** [comment]
thanks again @OlkaA. I did that.

**Pull request merged**